### PR TITLE
Style context

### PR DESCRIPTION
**Pull request by jquense**
w/d/y/t ?

### CODE REVIEW

File: src/Button.js [hunk 1]
```diff
@@ -3,100 +3,105 @@ import React from 'react';
...
+      className,
+      block,
+      type,
+      componentClass,
+      ...props
+    } = this.props;
 
-    if (elementProps.href) {
-      return this.renderAnchor(elementProps, fullClassName);
+    delete props.bsRole;
```
Comment (taion) [inline src/Button.js]: can we do something like `bsRole: _0` in the spread above, and add a lint rule to not warn on unused vars starting with `_`? i think that's what we're largely doing now, and i think it's good

File: src/Button.js [hunk 2]
```diff
@@ -3,100 +3,105 @@ import React from 'react';
...
+
+    const classes = classNames(
+      className,
+      bsPrefix,
+      active && 'active',
+      `${bsPrefix}-${variant}`,
```
Comment (taion) [inline src/Button.js]: do we want to support the null `bsPrefix` case?

File: src/ThemeProvider.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      </ThemeContext.Provider>
+    );
+  }
+}
+
+function createBootstrapComponent(Component, prefix) {
+  const name = Component.displayName || Component.name;
+  // eslint-disable-next-line
+  return class extends React.Component {
+    static displayName = `Boostrap(${name})`;
```
Comment (taion) [inline src/ThemeProvider.js]: "Bootstrap"

File: src/Button.js (hunk 2 above)
Comment (jquense) [reply src/Button.js]: I think no. I couldn't think of any cases where it'd make sense. I guess if you want a completely unstyled component

**Pull request merged**